devtools/gradle/gradle-application-plugin/src/test/java/io/quarkus/gradle/tooling/ConditionalDependenciesEnablerTest.java

```diff
@@ -0,0 +1,306 @@
+        "dependencies {\n" +
+        "\timplementation platform(\"io.quarkus:quarkus-bom:${quarkusVersion}\")\n" +
+        "}\n" +
+        "publishing {\n" +
+        "\tpublications {\n" +
+        "\t\tmavenJava(MavenPublication) {\n" +
+        "\t\t\tfrom components.java\n" +
+        "\t\t}\n" +
+        "\t}\n" +
+        "}";
+    Files.writeString(
+        testProjectDir.resolve(Path.of("extension-a", "build.gradle")),
+        extension_a_build_gradle,
+        StandardOpenOption.CREATE_NEW
+    );
+
+    //language=Groovy
+    String extension_b_build_gradle =
+        "plugins {\n" +
+        "\tid 'java-library'\n" +
+        "\tid 'maven-publish'\n" +
+        "\tid 'io.quarkus.extension'\n" +
+        "}\n" +
+        "repositories {\n" +
+        "\tmavenCentral()\n" +
+        "\tmavenLocal()\n" +
+        "}\n" +
+        "group 'org.test'\n" +
+        "version '1.0-SNAPSHOT'\n" +
+        "evaluationDependsOn(':extension-b-deployment')\n" +
+        "quarkusExtension {\n" +
+        "\tdeploymentModule = 'extension-b-deployment'\n" +
+        "\tdependencyConditions = ['org.test:extension-c']\n" +
+        "}\n" +
+        "dependencies {\n" +
+        "\timplementation platform(\"io.quarkus:quarkus-bom:${quarkusVersion}\")\n" +
+        "}\n" +
+        "publishing {\n" +
+        "\tpublications {\n" +
+        "\t\tmavenJava(MavenPublication) {\n" +
+        "\t\t\tfrom components.java\n" +
+        "\t\t}\n" +
+        "\t}\n" +
+        "}";
+    Files.writeString(
+        testProjectDir.resolve(Path.of("extension-b", "build.gradle")),
+        extension_b_build_gradle,
+        StandardOpenOption.CREATE_NEW
+    );
+
+    //language=Groovy
+    String extension_c_build_gradle =
+        "plugins {\n" +
+        "\tid 'java-library'\n" +
+        "\tid 'maven-publish'\n" +
+        "\tid 'io.quarkus.extension'\n" +
+        "}\n" +
+        "repositories {\n" +
+        "\tmavenCentral()\n" +
+        "\tmavenLocal()\n" +
+        "}\n" +
+        "group 'org.test'\n" +
+        "version '1.0-SNAPSHOT'\n" +
+        "evaluationDependsOn(':extension-c-deployment')\n" +
+        "quarkusExtension {\n" +
+        "\tdeploymentModule = 'extension-c-deployment'\n" +
+        "}\n" +
+        "dependencies {\n" +
+        "\timplementation platform(\"io.quarkus:quarkus-bom:${quarkusVersion}\")\n" +
+        "}\n" +
+        "publishing {\n" +
+        "\tpublications {\n" +
+        "\t\tmavenJava(MavenPublication) {\n" +
+        "\t\t\tfrom components.java\n" +
+        "\t\t}\n" +
+        "\t}\n" +
+        "}";
+    Files.writeString(
+        testProjectDir.resolve(Path.of("extension-c", "build.gradle")),
+        extension_c_build_gradle,
+        StandardOpenOption.CREATE_NEW
+    );
+    BuildResult result = GradleRunner.create()
+        .withProjectDir(testProjectDir.toFile())
+        .withArguments("publishToMavenLocal")
+        .withPluginClasspath()
+        .build();
```

Comment:
This is actually working around a potential bug in which the plugin code cannot resolve conditional extensions which are not published. The integration test was written against 2.8.0.Final vanilla before I did any refactoring, so the issue is present in released Quarkus. I can file an issue if this is not intended behavior.

Reply:
Oh this is perfect! Awesome. I don't mind the tests being a little crusty as long as they get the testing done :) I didn't even think to look beyond the devtools directory. Thank you for the references.

Reply:
I've removed these tests as the ones you linked cover it and more.